lib/rbac/filterer.rb

```diff
@@ -326,6 +329,25 @@ def inline_view?(options, scope)
         scope.respond_to?(:includes_values)
     end
 
+    # Convert an order by column to the select columns
+    # @param [Array] columns the order by clause
+    # @return [Array] columns useable in the select clause
+    #
+    # this is similar to Connection#columns_for_distinct
+    #
+    # For a query with a distinct, all columns in the order by clause
+    # need to be in the select clause. This method gives us the columns for the select
+    # we currently assume all regular columns are already in the select. we just need the functions
+    def select_from_order_columns(columns)
+      columns.map do |column|
+        column.kind_of?(Arel::Nodes::Ordering) ? column.expr : column
+      end.reject do |column|
+        column.kind_of?(Arel::Attributes::Attribute) ||
+          column.kind_of?(Symbol) ||
+          (column.kind_of?(String) && column =~ /^("?[a-z_0-9]+"?[.])"?[a-z_0-9]+"? ?(DESC|ASC)?$/i)
```

Comment:
😮

Reply:
@kbrock what is the return of this method for your example tests?

Based on your comment, it looks like it discards all order by columns... what exactly become `extra_cols` in `apply_select`?

Reply:
Do we need separate tests for this method so I can have a chance to understand this?  🤣

Reply:
added to the comments and added tests (thanks, this found a bug